pkg/volume/git_repo/git_repo_test.go

```diff
@@ -55,106 +56,408 @@ func TestCanSupport(t *testing.T) {
 	}
 }
 
-func testSetUp(plug volume.VolumePlugin, builder volume.Builder, t *testing.T) {
-	var fcmd exec.FakeCmd
-	fcmd = exec.FakeCmd{
-		CombinedOutputScript: []exec.FakeCombinedOutputAction{
-			// git clone
-			func() ([]byte, error) {
-				os.MkdirAll(path.Join(fcmd.Dirs[0], "kubernetes"), 0750)
-				return []byte{}, nil
+// Spec for plugin
+type pluginSpec struct {
+	pluginName string
+	volPath    string
+}
+
+// Expected command
+type expectedCommand struct {
+	// The git command
+	cmd []string
+	// The dir of git command is executed
+	dir string
+}
+
+func TestPlugin(t *testing.T) {
+	gitUrl := "https://github.com/GoogleCloudPlatform/kubernetes.git"
+	revision := "2a30ce65c5ab586b98916d83385c5983edd353a1"
+	validPluginSpec := pluginSpec{
+		volPath:    "pods/poduid/volumes/kubernetes.io~git-repo/vol1",
+		pluginName: "kubernetes.io/git-repo",
+	}
+
+	scenarios := map[string]struct {
```

Comment:
Nit: I personally prefer to use an array here like so:

```go
cases := []struct{
  name string
  // other fields
}
```

...because:

1.  iteration order through maps is not deterministic
2.  it helps keep indentation in check